File: RedSheet.DbEntities/Models/CultureSummary.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("CultureSummary", Schema = "dbo")]
	[Serializable]
	public partial class CultureSummary  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CultureSummaryId { get; set; } 

	public string CultureSummaryText { get; set; } 
 
	[Required]
	public bool OurEgalitarian { get; set; } 
 
	[Required]
	public bool OurIndividualistic { get; set; } 
 
	[Required]
	public bool OurMonochronic { get; set; } 
 
	[Required]
	public bool OurShortTerm { get; set; } 
 
	[Required]
	public bool TheirEgalitarian { get; set; } 
 
	[Required]
	public bool TheirIndividualistic { get; set; } 
 
	[Required]
	public bool TheirMonochronic { get; set; } 
 
	[Required]
	public bool TheirShortTerm { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	//Relationships
	public int OurCultureCategoryId { get; set; } 

	public ApplicationObject ApplicationObject { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	//Relationships
	public int TheirCultureCategoryId { get; set; } 

	public ApplicationObject ApplicationObject1 { get; set; } 

	public CultureSummary()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/culture-summary.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ApplicationObject,  } from './'
export class CultureSummary {
    constructor(cultureSummary?: CultureSummary )  {
        let properties = [ "cultureSummaryId", "cultureSummaryText", "ourEgalitarian", "ourIndividualistic", "ourMonochronic", "ourShortTerm", "theirEgalitarian", "theirIndividualistic", "theirMonochronic", "theirShortTerm", "ourCultureCategoryId", "theirCultureCategoryId",];
        for (let property of properties)
            if (cultureSummary && cultureSummary[property])
                this[property] = cultureSummary[property];
    }
 
	cultureSummaryId : number =   0 ;
 
	cultureSummaryText : string =   undefined;
 
	ourEgalitarian : boolean = false ;
 
	ourIndividualistic : boolean = false ;
 
	ourMonochronic : boolean = false ;
 
	ourShortTerm : boolean = false ;
 
	theirEgalitarian : boolean = false ;
 
	theirIndividualistic : boolean = false ;
 
	theirMonochronic : boolean = false ;
 
	theirShortTerm : boolean = false ;
 
    @range(0,2147483647)
	ourCultureCategoryId : number =   undefined;
	applicationObject : ApplicationObject  ;
 
    @range(0,2147483647)
	theirCultureCategoryId : number =   undefined;
	applicationObject1 : ApplicationObject  ;


}
